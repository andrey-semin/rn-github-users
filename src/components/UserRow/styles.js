import { StyleSheet } from 'react-native'
import { dimensions } from 'styles'

export default StyleSheet.create({
  root: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingVertical: 10,
    paddingHorizontal: dimensions.gutter
  },
  avatar: {
    height: 100,
    width: 100,
    borderRadius: 50
  },
  userConatiner: {
    paddingLeft: dimensions.gutter
  }
})
