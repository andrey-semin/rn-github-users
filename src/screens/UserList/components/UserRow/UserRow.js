import React from 'react'
import PropTypes from 'prop-types'
import { View, Text, Image, TouchableOpacity } from 'react-native'

import styles from './styles'

const UserRow = ({ user, onPress }) => (
  <TouchableOpacity onPress={onPress}>
    <View style={styles.root}>
      <Image source={{ uri: user.avatar_url }} style={styles.avatar} />
      <View style={styles.userConatiner}>
        <Text>{user.login}</Text>
        <Text>{user.html_url}</Text>
      </View>
    </View>
  </TouchableOpacity>
)
UserRow.propTypes = {
  user: PropTypes.shape({
    avatar_url: PropTypes.string.isRequired,
    login: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired
  }),
  onPress: PropTypes.func.isRequired
}

export default UserRow
