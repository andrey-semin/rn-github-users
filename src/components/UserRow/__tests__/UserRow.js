import React from 'react'
import renderer from 'react-test-renderer'

import UserRow from '../UserRow'

it('UserRow renders correctly', () => {
  const tree = renderer
    .create(
      <UserRow
        user={{
          avatar_url: 'some url',
          login: 'big man',
          html_url: 'some html url'
        }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
