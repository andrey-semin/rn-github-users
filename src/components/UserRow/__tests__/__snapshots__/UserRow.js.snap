// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRow renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "some url",
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
    <View
      style={
        Object {
          "paddingLeft": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        big man
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        some html url
      </Text>
    </View>
  </View>
</View>
`;
